Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeyTable+execute.swift

```diff
@@ -0,0 +1,217 @@
+                                           additionalWhereClause: additionalWhereClause,
+                                           nextToken: returnedNextToken)
+                    .map { partialResult in
+                        // return the results from 'this' call and all later paginated calls
+                        return paginatedItems.0 + partialResult
+                    }
+            } else {
+                // this is it, all results have been obtained
+                let promise = self.eventLoop.makePromise(of: [ReturnedType].self)
+                promise.succeed(paginatedItems.0)
+                return promise.futureResult
+            }
+        }
+    }
+    
+    private func monomorphicPartialExecute<AttributesType, ItemType>(
+            partitionKeys: [String],
+            attributesFilter: [String]?,
+            additionalWhereClause: String?,
+            nextToken: String?) -> EventLoopFuture<[TypedDatabaseItem<AttributesType, ItemType>]> {
+        let executeFuture: EventLoopFuture<([TypedDatabaseItem<AttributesType, ItemType>], String?)> =
+            monomorphicExecute(partitionKeys: partitionKeys,
+                               attributesFilter: attributesFilter,
+                               additionalWhereClause: additionalWhereClause,
+                               nextToken: nextToken)
+        
+        return executeFuture.flatMap { paginatedItems in
+            // if there are more items
+            if let returnedNextToken = paginatedItems.1 {
+                // returns a future with all the results from all later paginated calls
+                return self.monomorphicPartialExecute(partitionKeys: partitionKeys,
+                                                      attributesFilter: attributesFilter,
+                                                      additionalWhereClause: additionalWhereClause,
+                                                      nextToken: returnedNextToken)
+                    .map { partialResult in
+                        // return the results from 'this' call and all later paginated calls
+                        return paginatedItems.0 + partialResult
+                    }
+            } else {
+                // this is it, all results have been obtained
+                let promise = self.eventLoop.makePromise(of: [TypedDatabaseItem<AttributesType, ItemType>].self)
+                promise.succeed(paginatedItems.0)
+                return promise.futureResult
+            }
+        }
+    }
+    
+    private func getStatement(partitionKeys: [String],
+                              attributesFilter: [String]?,
+                              partitionKeyAttributeName: String,
+                              additionalWhereClause: String?) -> String {
+        let attributesFilterString = attributesFilter?.joined(separator: ", ") ?? "*"
+        
+        let partitionWhereClause: String
+        if partitionKeys.count == 1 {
```

Comment:
Should we consider the case where `partitionKeys` list is empty?

Reply:
Good catch, I have added guard statements before this function is called to not even make a dynamodb call if there are no partition keys.